src/extension.ts

```diff
@@ -0,0 +1,41 @@
+import * as vscode from 'vscode';
+import { exec } from 'child_process';
+
+export function activate(context: vscode.ExtensionContext) {
+	var installed: Boolean = false;
+	//check if alloy is installed
+	exec("alloy --version", (error, stdout, stderr) => {
+		if (error || stderr) {
+			vscode.window.showWarningMessage(`Alloy is not installed. Please install it from https://grafana.com/docs/alloy/latest/set-up/install/`);
+		}
+		if (stdout) {
+			vscode.window.showInformationMessage(`Alloy is installed: ${stdout}`);
```

Comment:
This would show the info message every time the extension is activated for the first time each session, right? I wonder if that might be too noisy; do other extensions do this?

Reply:
I have no idea, this is my first time writing an extension. I will say, that whenever I startup a new session, I get all sorts of messages, but I'm not sure.

Reply:
Happy to do whatever you think is best, there are very few "simple" extensions that have good best practices for me to copy, so much of this ecosystem is layered jank.

Reply:
It's something we can always change again later, I'm fine leaving it as-is for now